### PR TITLE
Bump fluent/fluent-logger-golang v1.4.0

### PR DESCRIPTION
**Pull request by thaJeztah**
- Add RequestAck to enable at-least-once message transferring
- Add Async option to update sending message in asynchronous way
- Deprecate AsyncConnect (Use Async instead)

full diff: https://github.com/fluent/fluent-logger-golang/compare/v1.3.0...v1.4.0

Also updating tinylib/msgp v1.1.0

full diff: https://github.com/tinylib/msgp/compare/3b556c64540842d4f82967be066a7f7fffc3adad...v1.1.0

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @tagomoris ptal 🤗

**Comment by thaJeztah** [comment]
Some flakiness in docker-py tests;

```
02:22:31 _______________________ ExecTest.test_exec_command_demux _______________________
02:22:31 /docker-py/tests/integration/api_exec_test.py:182: in test_exec_command_demux
02:22:31     assert next(exec_log) == (b'hello out\r\n', None)
02:22:31 E   AssertionError: assert ('hello out\r...rr\r\n', None) == ('hello out\r\n', None)
02:22:31 E     At index 0 diff: 'hello out\r\nhello err\r\n' != 'hello out\r\n'
02:22:31 E     Use -v to get the full diff
02:22:31 - generated xml file: /go/src/github.com/docker/docker/bundles/test-docker-py/results.xml -
02:22:31 =========================== short test summary info ============================
```

**Review by cpuguy83 — APPROVED** [review]
LGTM

File: daemon/logger/fluentd/fluentd.go [hunk 1]
```diff
@@ -111,9 +111,9 @@ func New(info logger.Info) (logger.Logger, error) {
 		maxRetries = int(mr64)
 	}
 
-	asyncConnect := false
+	async := false
 	if info.Config[asyncConnectKey] != "" {
-		if asyncConnect, err = strconv.ParseBool(info.Config[asyncConnectKey]); err != nil {
+		if async, err = strconv.ParseBool(info.Config[asyncConnectKey]); err != nil {
```
Comment (tagomoris) [inline daemon/logger/fluentd/fluentd.go]: IMO it's better to do:
* add `asyncKey` (fluentd-async) to enable `async` option
* warn that `asyncConnectKey` is now not recommended and will be removed later, and also enable `async` if `asyncConnectKey` is specified (and unless `asyncKey` is disabled explicitly)

**Review by tagomoris — COMMENTED** [review]
I couldn't find any code to set `RequestAck`.
Does this change include it?

File: daemon/logger/fluentd/fluentd.go (hunk 1 above)
Comment (thaJeztah) [reply daemon/logger/fluentd/fluentd.go]: Ah, right; my main goal with this PR was to keep the dependency up-to-date with upstream fixes, without changing the UX / options.

I'd be ok with adding a `fluentd-async` and marking `fluentd-async-connect` as deprecated (and perhaps adding an error or warning if both are specified?)

Probably better for a follow-up 🤗

**Comment by thaJeztah** [comment]
> I couldn't find any code to set RequestAck. Does this change include it?

No it doesn't (see my other comment https://github.com/moby/moby/pull/39074#discussion_r275694848); I can add it in that follow-up (it's a boolean opt, right?)

File: daemon/logger/fluentd/fluentd.go (hunk 1 above)
Comment (thaJeztah) [reply daemon/logger/fluentd/fluentd.go]: Let me rename the variable back in this PR though

**Comment by tagomoris** [comment]
I see. LGTM.

**Comment by thaJeztah** [comment]
Opened a follow-up PR with user-facing changes and deprecation in https://github.com/moby/moby/pull/39086

**Review by yongtang — APPROVED** [review]
LGTM

**Pull request merged**